src/lib/y2storage/arch.rb

```diff
@@ -53,6 +57,46 @@ class Arch
     #   @return [Integer] the system page size
     storage_forward :page_size
 
+    # @!method self.efibootmgr?
+    #
+    # @return [Boolean] whether the UEFI boot manager can write UEFI boot entries
+    storage_class_forward :efibootmgr?
+
+    # @!method efibootmgr?
+    #
+    # Whether the UEFI boot manager can write UEFI boot entries.
+    #
+    # Storage::Arch exports it as static function to not break the existing ABI.
+    # Make it also available as instance method to have a consistent API
+    # in {Y2Storage::Arch}.
+    #
+    # @note This is entirely independent of {#efiboot?}. In particular
+    #   overriding {#efiboot?} with environment variables or config files will
+    #   have no effect on {#efibootmgr?}.
+    #
+    # @return [Boolean] whether the UEFI boot manager can write UEFI boot entries
+    def efibootmgr?
+      self.class.efibootmgr?
+    end
+
+    # Wraps the Storage::Arch object passed to it or creates a new
+    # Storage::Arch object and wraps that.
+    #
+    # This also adjusts {#efiboot?} according to the `/etc/install.inf::EFI` setting.
+    #
+    # @param storage_arch [Storage::Arch] Storage::Arch object to wrap.
+    #
+    # @return [Y2Storage::Arch]
+    def self.create(storage_arch = Storage::Arch.new)
```

Comment:
NP: I would avoid having both: `#new` and `#create`. The only difference is whether to adjust *efiboot* or not. What about overloading the constructor?

~~~
def initialize(storage_arch = Storage::Arch.new, adjust_efiboot: true)
  super(storage_arch)

  return unless adjust_efiboot
  return uless Yast::Linuxrc.InstallInf("EFI")
  
  storage_arch.efiboot = Yast::Linuxrc.InstallInf("EFI") == "1"  
end
~~~

Reply:
There's no point in making the config adjustment optional. You'd just forcibly allow the `efi` boot setting and `LIBSTORAGE_EFI` to diverge. And that's what we want to avoid.